Assets/Scripts/Control&Input/ItemSelectSlot.cs

```diff
@@ -66,20 +64,32 @@ public void SetItems(List<Item> items, Item initialItem)
         for (var i = 0; i < items.Count; i++)
         {
             // Item is probably an empty extension slot!
-            if (items[i] == null) 
+            if (items[i] == null)
             {
                 var missing = Instantiate(missingObject, Vector3.zero, itemHolder.rotation, itemHolder);
                 missing.transform.localPosition = Vector3.zero;
-                augmentModels.Add((missing, missing.transform.localPosition));
+                augmentModels.Add((missing, missing.transform.localPosition, Vector3.zero));
                 continue;
             }
+
+            // Item is an augment
             var rotation = Quaternion.Euler(new Vector3(0, 90, -20));
             var instance = Instantiate(items[i].augment, Vector3.zero, rotation, itemHolder);
             instance.transform.ScaleAndParent(itemModelScale, itemHolder);
-            instance.transform.localPosition = itemSlots[i].transform.position;
+            Transform itemCenter;
+            if (items[i].augmentType == AugmentType.Body)
+            {
+                itemCenter = instance.GetComponent<GunBody>().midpoint;
```

Comment:
Ye, I definetly think gunbody should inherit from augment, but eh, we can refactor later